### PR TITLE
Add makeFixedOutputPath

### PR DESCRIPTION
**Pull request by puffnfresh**

### CODE REVIEW

**Comment by puffnfresh** [comment]
> Why do we special case this to SHA256?

At the moment `makeStorePath` is special cased to SHA256, which the function uses.

I've fixed the digest problem. Seems to be working.

**Comment by shlevy** [comment]
Ah can you rebase on/merge in https://github.com/haskell-nix/hnix-store/pull/42 ?

File: hnix-store-core/src/System/Nix/ReadonlyStore.hs [hunk 1]
```diff
@@ -28,5 +28,14 @@ makeTextPath nm h refs = makeStorePath ty h nm
   where
     ty = BS.intercalate ":" ("text" : map storePathToRawFilePath (HS.toList refs))
 
+makeFixedOutputPath :: (KnownStoreDir storeDir, ValidAlgo hashAlgo, NamedAlgo hashAlgo) => Bool -> Digest hashAlgo -> StorePathName -> StorePath storeDir
+makeFixedOutputPath recursive h nm =
+  makeStorePath ty h' nm
+  where
+    (ty, h') =
+      if recursive
```
Comment (shlevy) [inline hnix-store-core/src/System/Nix/ReadonlyStore.hs]: Only if recursive and sha256, right?

Comment (shlevy) [reply hnix-store-core/src/System/Nix/ReadonlyStore.hs]: https://github.com/NixOS/nix/blob/41ba5135e0a2d9940c4e8cbedff44eb1a47af782/src/libstore/store-api.cc#L200

Comment (puffnfresh) [reply hnix-store-core/src/System/Nix/ReadonlyStore.hs]: Ah maybe that's why I had it fixed to SHA256 previously. Will update!

Comment (puffnfresh) [reply hnix-store-core/src/System/Nix/ReadonlyStore.hs]: @shlevy I've updated, it will now only use "source" if recursive and SHA256.

**Comment by Ericson2314** [comment]
We should add something about the references that are unstable in Nix, but this looks good to me.

**Pull request merged**